### PR TITLE
home work

### PR DESCRIPTION
**Pull request by drelv**

### CODE REVIEW

File: calculator.test.js [hunk 1]
```diff
@@ -6,16 +6,26 @@ var calculator = require('./calculator');
...
   // test of remainder
+  it('should get the remainder', function() {
+    console.log(calculator.remainder(9,2));
+  });
+
+    // add string
+  it('should get the sum from string', function() {
+    assert.strictEqual(calculator.addString("1,2,3,4"),10);
+  });
+
```
Comment (vudknguyen) [inline calculator.test.js:27]: missing test case(s) to verify addString will throw exception if a number in argument is negative

File: calculator.test.js [hunk 2]
```diff
@@ -6,16 +6,26 @@ var calculator = require('./calculator');
...
   it('should multiply two numbers', function() {
-    // your code here
+    assert.equal(calculator.multiply(2,2), 4);
   });
   
-  // test for divide
-  
   // test of remainder
+  it('should get the remainder', function() {
+    console.log(calculator.remainder(9,2));
```
Comment (vudknguyen) [inline calculator.test.js:20]: Should use assert for test framework able to detect fail test case(s)

File: calculator.test.js [hunk 3]
```diff
@@ -6,16 +6,26 @@ var calculator = require('./calculator');
...
-  it('should add two numbers', function() {
-    assert.equal(calculator.add(1,1), 2);
+  it('should add numbers', function() {
+    assert.equal(calculator.add(2,2,2,2,2), 10); // test will fail
   });
   
+  
   it('should multiply two numbers', function() {
-    // your code here
+    assert.equal(calculator.multiply(2,2), 4);
```
Comment (vudknguyen) [inline calculator.test.js:15]: Should avoid this special case because 2 + 2 = 4 as well

File: calculator.js [hunk 4]
```diff
@@ -1,13 +1,30 @@
...
-
 // bahavior for remainder
+exports.remainder = function(val1, val2) {
+  return Number(val1) % Number(val2);
+}
+
+exports.addString = function(val) {
+	var qc = val.split(",");
+	var sum=0;
+	if(val.indexOf("-1")>-1) throw "negatives not allowed";
```
Comment (vudknguyen) [inline calculator.js]: What if the number is negative other than -1 ?

File: calculator.test.js [hunk 5]
```diff
@@ -6,16 +6,26 @@ var calculator = require('./calculator');
 
 describe('Calculator', function() {
   
-  it('should add two numbers', function() {
-    assert.equal(calculator.add(1,1), 2);
+  it('should add numbers', function() {
+    assert.equal(calculator.add(2,2,2,2,2), 10); // test will fail
```
Comment (vudknguyen) [inline calculator.test.js:10]: Accordingly to comment, test will fail ?, I think it should pass to verify implement correctly ?

Comment (vudknguyen) [inline calculator.test.js:10]: How about test case for add 2 numbers, does this function work well with add 2 numbers ?

Comment (drelv) [inline calculator.test.js:10]: HI vu, what i mean on this is that the method was not yet updated to handle the multiple arguments. 

```
On Friday, June 24, 2016 12:27 PM, Vu Nguyen <notifications@github.com> wrote:
```

 In calculator.test.js:> @@ -6,16 +6,26 @@ var calculator = require('./calculator');

File: calculator.js (hunk 4 above)
Comment (drelv) [inline calculator.js]: Should something like this.
for(var i=0;i<qc.length;i++){
        var n = Number(qc[i]);
        if(n<0) throw "negatives not allowed";
        sum+= Number(qc[i]); 
    }

File: calculator.test.js (hunk 5 above)
Comment (drelv) [inline calculator.test.js:10]: and yes then forgot to remove the comment.

Comment (drelv) [inline calculator.test.js:10]: > > How about test case for add 2 numbers, does this function work well with add 2 numbers ? . the improved add method should already handle this.

File: calculator.test.js (hunk 2 above)
Comment (drelv) [inline calculator.test.js:20]: i was thinking that the test case for this is to test if the method already exist
